napari/_qt/util.py

```diff
@@ -25,12 +43,8 @@ def create_qt_properties(layer):
         properties : napari.layers.base.QtLayerProperties
             Qt propetry widget
     """
-    name = 'Qt' + type(layer).__name__ + 'Properties'
-    try:
-        qt_props = globals()[name]
-        properties = qt_props(layer)
-    except KeyError:
-        properties = QtLayerProperties(layer)
+    name = type(layer).__name__
+    properties = layer_to_properties[name](layer)
```

Comment:
I would make the keys layers instead of their names

Reply:
additionally, you seem to re-use this functionality many times so it may be better just to create a more general function

Reply:
something that could be cool that you might consider is using `__init_subclass__` in combination with `__new__`:
```python
class QtLayerProperty:
    _layer_properties = {}

    def __new__(cls, layer):
        if cls is QtLayerProperty and type(layer) in QtLayerProperty._layer_properties:
            cls = QtLayerProperty._layer_properties[type(layer)]
        return object.__new__(cls)

    def __init_subclass__(cls, layer=None, **kwargs):
        if issubclass(layer, Layer):
            QtLayerProperty._layer_properties[layer] = cls
        
        super().__init_subclass__(**kwargs)

class QtImageProperty(QtLayerProperty, layer=Image):
    ...

>>> QtLayerProperty(viewer.layers['Image 0'])
QtImageProperty(...)
```

@sofroniewn @jni thoughts?

Reply:
this look very cool - but can you make this in a follow up PR instead?

Reply:
could you at least address the first comment then?

Reply:
i've switched over to making the keys layers - good suggestion. i also do like the proposed scheme a lot too - it would mean we don't need these weird convertors in utils. if @jni wants i can try and add now too, but also waiting to a follow-up PR is fine for me too as I said above.

Reply:
@kne42 I tired making this change myself, but ran into errors (probably easily fixable) but I don't fully understand it, so I'll leave it to you!